modules/reactor/src/vectorpostprocessors/IntegralReportingIDVPP.C

```diff
@@ -0,0 +1,165 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "IntegralReportingIDVPP.h"
+
+#include "MooseVariable.h"
+#include "MooseMesh.h"
+
+#include "libmesh/mesh_base.h"
+
+registerMooseObject("ReactorApp", IntegralReportingIDVPP);
+
+InputParameters
+IntegralReportingIDVPP::validParams()
+{
+  InputParameters params = ElementVariableVectorPostprocessor::validParams();
+  params.addRequiredParam<std::vector<ExtraElementIDName>>(
+      "reporting_id", "The types of reporting id by which to separate integrals.");
+  params.addClassDescription(
+      "This IntegralReportingIDVPP source code is to integrate variables based on reporting IDs");
+  return params;
+}
+
+IntegralReportingIDVPP::IntegralReportingIDVPP(const InputParameters & parameters)
+  : ElementVariableVectorPostprocessor(parameters),
+    _nvar(coupledComponents("variable")),
+    _reporting_id(getParam<std::vector<ExtraElementIDName>>("reporting_id")),
+    _n_reporting_id(_reporting_id.size())
+{
+  // numbers of parsed vpp ids
+  dof_id_type n_vpp_ids;
+  // list of extra_integer id
+  std::set<dof_id_type> ids;
+
+  std::set<ExtraElementIDName> _reporting_id_copy(_reporting_id.begin(), _reporting_id.end());
```

Comment:
```suggestion
  std::set<ExtraElementIDName> reporting_id_set(_reporting_id.begin(), _reporting_id.end());
```

`_` prefix is reserved for member vars